### PR TITLE
Complex Multigeometries In/Out 

### PR DESCRIPTION
**Pull request by zjiang0529**
Hi! We have pushed a first version of adding multi-geometries features.
(Issue #109) 
Thanks.

### CODE REVIEW

File: pygem/nurbshandler.py [hunk 1]
```diff
@@ -7,15 +7,24 @@
...
+from OCC.ShapeFix import (ShapeFix_ShapeTolerance, ShapeFix_Shell)
+from OCC.ShapeAnalysis import ShapeAnalysis_WireOrder
 from OCC.StlAPI import StlAPI_Writer
-from OCC.TopAbs import (TopAbs_FACE, TopAbs_EDGE)
-from OCC.TopExp import TopExp_Explorer
+from OCC.TopAbs import (TopAbs_FACE, TopAbs_EDGE, TopAbs_WIRE,
+						TopAbs_FORWARD, TopAbs_SHELL)
+from OCC.TopExp import (TopExp_Explorer, topexp)
 from OCC.gp import (gp_Pnt, gp_XYZ)
+from OCC.BRep import *
```
Comment (mtezzele) [inline pygem/nurbshandler.py]: See quantified code for suggestion on how to avoid the import using *

File: pygem/nurbshandler.py [hunk 2]
```diff
@@ -7,15 +7,24 @@
...
+from OCC.TColgp import (TColgp_Array1OfPnt, TColgp_Array2OfPnt)
+from OCC.BRepOffsetAPI import *
```
Comment (mtezzele) [inline pygem/nurbshandler.py]: Same as above

File: pygem/nurbshandler.py [hunk 3]
```diff
@@ -205,20 +222,386 @@ def write(self, mesh_points, filename, tolerance=None):
...
-			brep_surf = BRepBuilderAPI_MakeFace(occ_face.GetHandle(), wire).Shape()
+			brep_surf = BRepBuilderAPI_MakeFace(occ_face.GetHandle(),
+												wire).Shape()
 			compound_builder.Add(compound, brep_surf)
 			n_faces += 1
 			faces_explorer.Next()
 		self.write_shape_to_file(compound, self.outfile)
 
+
+	def parse_face(self, topo_face):
```
Comment (mtezzele) [inline pygem/nurbshandler.py]: You never use self in this method. Make it a static method (see quantified code). Also the methods below have the same problem.

File: pygem/nurbshandler.py [hunk 4]
```diff
@@ -205,20 +222,386 @@ def write(self, mesh_points, filename, tolerance=None):
...
+		Edges of the Face.
+
+		:param topo_face: the input Face
+
+		:return: mesh_points_face: it is a `n_points`-by-3 matrix containing the
+		coordinates of the control points of the Face (a nurbs surface)
+
+		:return: mesh_points_edge: it is a list of `n_points`-by-3 matrix
+
+		:rtype: numpy.ndarray and list
```
Comment (mtezzele) [inline pygem/nurbshandler.py]: Actually it is a dictionary

File: pygem/nurbshandler.py [hunk 5]
```diff
@@ -205,20 +222,386 @@ def write(self, mesh_points, filename, tolerance=None):
...
+	def parse_face(self, topo_face):
+		"""
+		Method to parse a single Face (a single patch nurbs surface).
+		It returns a matrix with all the coordinates of control points of the
+		Face and a second list with all the control points related to the
+		Edges of the Face.
+
+		:param topo_face: the input Face
+
+		:return: mesh_points_face: it is a `n_points`-by-3 matrix containing the
```
Comment (mtezzele) [inline pygem/nurbshandler.py]: You can merge these two returns by creating a single variable that will be returned. Please check also the other methods.

**Comment by mtezzele** [comment]
Any news @zjiang0529 ?

**Comment by zjiang0529** [comment]
Hello, thanks for the remind. We are going to commit a new version soon.

**Comment by zjiang0529** [comment]
Hello, we have just committed some modification. 
Now we think it would be time to discuss how to integrate and merge 
with the parse() and write() functions in Nurbshandler class.

**Comment by mtezzele** [comment]
@zjiang0529 please add some tests in order to verify that everything is working fine.

**Pull request closed without merging**